spark/src/main/scala/org/allenai/pipeline/spark/PartitionedRddArtifact.scala

```diff
@@ -0,0 +1,9 @@
+package org.allenai.pipeline.spark
+
+import org.allenai.pipeline._
+
+trait PartitionedRddArtifact extends Artifact {
```

Comment:
This Artifact class is a common interface for saving RDDs. Previously, we had separate implementations for File- and S3-based artifacts, which required some ugly run-time dynamic casting.